First adaptations (please see comments)

jstrnbrg opened the pull request:
unfortunately all earlier git commit messages got lost when setting this up.
I will try to summarise what I did in the comments here.

jstrnbrg commented:
saleor/account/forms.py: Shows email field instead of just name

jstrnbrg commented:
saleor/core/urls.py: Adds reseller routing

jstrnbrg commented:
saleor/core/views.py: reseller view is only accessible if user is logged in, if user is not, it will redirect to home view

jstrnbrg commented:
saleor/payment/gateways/globee/init.py: Playing around with Globee integration

jstrnbrg commented:
saleor/secrets.py: File to store secrets. File is gitignored. When using docker these variables will need to be specified again in the common.env 
I'm running it locally without docker. 'python3 manage.py runserver'

jstrnbrg commented:
saleor/settings.py: Importing the secret variables

Tomasvrba commented on CHANGELOG.md:
```diff
@@ -3,72 +3,59 @@
+  - Estonian
+  - Indonesian
+- Fix problem with l10n in Braintree payment gateway template - #3691 by @Kwaidan00
+- Improve vouchers country limiting  - #3707 by @michaljelonek
```
Any idea where all the changes to CHANGELOG.md are coming from? I don't think anything should have changed in there?
Maybe since you originally pulled from https://github.com/mirumee/saleor, that repo could be a little different now from our fork.
Pretty minor, but might want to make sure this file is tracking the official CHANGELOG

jstrnbrg replied:
Yeah I assume all these other changes were made to the master repo within the 3 days from when I cloned and now pushed. 
Does that make sense?

jstrnbrg commented:
saleor/settings.py:
- Changed 'DEFAULT_COUNTRY' to CH
- Changed 'DEFAULT_CURRENCY' to CHF
- AVAILABLE_CURRENCIES = [DEFAULT_CURRENCY, "EUR", "BTC"] //this won't work until OPENEXCHANGERATES_API_KEY is set
- Set LOW_STOCK_THRESHOLD = 100
- Set MAX_CART_LINE_QUANTIT to 10000 (for resellers)
- LOGOUT_ON_PASSWORD_CHANGE = True
- Enabled Stripe payments, in Tesmode currently, change keys in secrets.py in order to use normal mode

Tomasvrba commented on common.env:
```diff
@@ -6,3 +6,6 @@ CELERY_BROKER_URL=redis://redis:6379/1
 VATLAYER_ACCESS_KEY
+
+STRIPE_PUBLIC_KEY = pk_test_jQx3R356B16Ozb4A6YsTMH7H
+STRIPE_SECRET_KEY = sk_test_PvAdUpfBifatDSFsbsw6jeeq
```
You forgot to remove the secret keys from here so the whole secret file is a little pointless at the moment :D
You can remove them in a new commit and then using git fixup to make it look like it never happened to make your new commit part of the previous commit: 
https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html

jstrnbrg replied:
hhahahahahha f***k thanks, will remove in next commit

jstrnbrg replied:
done

jstrnbrg commented:
templates/base.html: 
- Changed "Register" button to "Reseller" and it will forward to the login page instead of register page
- Commented out the "Login" button
- Changed Saleor Logo to Shift Logo
- Commented out "Account" section in the footer
- Changed copyright in footer

Tomasvrba commented on saleor/account/urls.py:
```diff
@@ -4,6 +4,7 @@
+    
     url(r'^$', views.details, name='details'),
 
     url(r'^login/$', views.login, name='login'),
```
Can get rid of empty lines here

jstrnbrg replied:
true

Tomasvrba commented on saleor/account/views.py:
```diff
@@ -107,6 +107,7 @@ def get_or_process_name_form(request):
 
 
+
 @login_required
```
Should only have 1 empty line before a function definition according to PEP8 standard :)

jstrnbrg replied:
thanks ;)

jstrnbrg replied:
One before and one after? So always 2 empty between 2 functions? @Tomasvrba

Tomasvrba replied:
Both, always 1 empty between functions, 2 empty between classes

jstrnbrg commented:
templates/cart_dropdown.html:
 - Before: "Checkout" in basket sends you to login/register/continue without login page.
- After: "Chekout sends you straight to the shipping address form page (where you would have been sent after clicking "continue without login") as I assume we don't want our users to create accounts.

Tomasvrba commented on saleor/core/views.py:
```diff
@@ -26,6 +28,12 @@ def home(request):
+        return redirect(home)
+    return TemplateResponse(request, 'reseller.html')
+
+
```
remove empty lines, you can install a linter for your editor to help you catch formatting issues. Or install flake8 on the command line with `pip3 install flake8` and then run `flake8 <your folder>`

jstrnbrg replied:
thanks

jstrnbrg commented:
templates/dashboard/order/pdf/base_pdf.html: The template for the PDFs that are sent per email (invoice etc)
- Changed Logo to our logo
- Inserted our address
- Question: What else needs to go on these PDFs? Any numbers? Telephone? VAT-ID? etc

jstrnbrg commented:
templates/reseller.html: created template

Tomasvrba commented on saleor/graphql/translations/mutations.py:
```diff
@@ -193,8 +182,7 @@ class ShopSettingsTranslate(BaseMutation):
     class Arguments:
-        language_code = graphene.Argument(
-            LanguageCodeEnum,
+        language_code = graphene.String(
```
Just wondering where all of these changes to the graphql files are coming from. Could you take a look at it please?

jstrnbrg replied:
as discussed

Tomasvrba commented on saleor/payment/gateways/globee/init.py:
```diff
@@ -0,0 +1,10 @@
+
+headers = {
+    'Accept': 'application/json',
+    'X-AUTH-KEY': 'u3rkCp5LPQz1EkTaMpjZIvJKtAqvnG21',
```
Should this key also be secret?

jstrnbrg replied:
does not matter, just created a throwaway account to test. We would create a Shift account if we decide on using globee and then we can make sure the key stays private

Tomasvrba commented on saleor/secrets.py:
```diff
@@ -0,0 +1,2 @@
+STRIPE_PUBLIC_KEY = "pk_test_jQx3R356B16Ozb4A6YsTMH7H"
+STRIPE_SECRET_KEY = "sk_test_PvAdUpfBifatDSFsbsw6jeeq"
```
You committed the secrets.
Forgot to add them to the .gitignore I guess? :)

jstrnbrg replied:
damn haha ;)

jstrnbrg replied:
added

Tomasvrba commented on saleor/settings.py:
```diff
@@ -7,6 +7,7 @@
 from django.contrib.messages import constants as messages
 from django.utils.translation import gettext_lazy as _, pgettext_lazy
 from django_prices.templatetags.prices_i18n import get_currency_fraction
+from saleor.secrets import STRIPE_PUBLIC_KEY, STRIPE_SECRET_KEY
```
you could also do `import secrets` 
and then below do 
`'public_key': secrets.STRIPE_PUBLIC_KEY`
same for the secret key 
Your call

jstrnbrg replied:
it would still be import saleor.secrets, right?

Tomasvrba replied:
Yes, that should be right :)

Tomasvrba commented on saleor/settings.py:
```diff
@@ -334,7 +335,7 @@ def get_bool_from_env(name, default_value):
 # Enabling vat requires valid vatlayer access key.
 # If you are subscribed to a paid vatlayer plan, you can enable HTTPS.
-VATLAYER_ACCESS_KEY = os.environ.get('VATLAYER_ACCESS_KEY')
+VATLAYER_ACCESS_KEY = '537cd7196abde3035ae3fb8716a412a6'
```
Should this be secret?

jstrnbrg replied:
same here, also throwaway account

Tomasvrba commented on saleor/static/dashboard-next/types/globalTypes.ts:
```diff
@@ -1,5 +1,4 @@
 /* tslint:disable */
-/* eslint-disable */
```
All the above changes to .ts and .tsx files, did you make any of them or were they introduced in the copying process? They can/should probably be discarded if you didn't make them. 
Overall, should try to only have your own changes in the commit. My bad though, that was partly my mistake with how we copied all the files :)

Tomasvrba commented on tests/api/test_page.py:
```diff
@@ -1,8 +1,8 @@
 import json
 
+from django.utils.text import slugify
```
No need to move this line?

jstrnbrg replied:
don't think i touched /tests/ will remove line though

Tomasvrba commented:
> saleor/secrets.py: File to store secrets. File is gitignored. When using docker these variables will need to be specified again in the common.env
> I'm running it locally without docker. 'python3 manage.py runserver'

See my review, secrets are still in the commit in both files

Tomasvrba commented:
Nice!
